implementing-nosto/advanced-implementation/accessing-state-with-selectors.md

```diff
@@ -0,0 +1,61 @@
+# Accessing state with selectors
+
+In Nosto, accessing the app state is simple. `useAppState` hook provides you full access to the entire application state.
+
+{% code title="State example" %}
+```jsx
+import { useAppState } from '@nosto/preact'
+
+export default () => {
+    const state = useAppState()
+	...
+}
+```
+{% endcode %}
+
+While simple, this approach has a number of disadvantages. Namely, any change to the state will trigger a re-render on every component that uses the state; and it may be awkward to access data that does not reside in a top level property.
```

Comment:
Please try rephrasing this sentence 
_it may be awkward to access data that does not reside in a top level property._

Try to put it in a non-technical way that's easy to understand

Reply:
And end it with
_For better performance and usability, consider the `useAppStateSelector` hook in the next section_